swap/src/protocol/bob/event_loop.rs

```diff
@@ -164,22 +165,26 @@ impl EventLoop {
                         _ => {}
                     }
                 },
-                Some((request, responder)) = self.spot_price_requests.next().fuse() => {
+
+                // Handle to-be-sent requests for all our network protocols.
+                // Use `self.is_connected_to_alice` as a guard to "buffer" requests until we are connected.
+                Some((request, responder)) = self.spot_price_requests.next().fuse(), if self.is_connected_to_alice() => {
                     let id = self.swarm.spot_price.send_request(&self.alice_peer_id, request);
                     self.inflight_spot_price_requests.insert(id, responder);
                 },
-                Some(((), responder)) = self.quote_requests.next().fuse() => {
+                Some(((), responder)) = self.quote_requests.next().fuse(), if self.is_connected_to_alice() => {
```

Comment:
❓ To re-validate my understanding - please correct me if I'm wrong: The spot-price protocol is independent of the swap and just requests the current BTC price for one XMR. The quote requests returns the amount of XMR for a given amount of BTC and these amounts are used in the execution setup.

Reply:
Not quite:

Quote is independent from swap. You can make as many quote requests as you want.

Spot-price must happen as a pre-condition to the execution-setup and the amount of the spot price protocol will be used for the execution. Spot-price should really be merged into execution-setup ...